### PR TITLE
[TACHYON-1057] Fix bug in removeBlocks

### PR DESCRIPTION
**Pull request by calvinjia**
https://tachyon.atlassian.net/browse/TACHYON-1057

A cleaner solution may be a fix the MasterBlockInfo interface. Currently we cannot use an iterator to remove the elements since an unmodifiable collection is returned.

### CODE REVIEW

**Comment by gpang** [comment]
LGTM

File: servers/src/main/java/tachyon/master/block/BlockMaster.java [hunk 1]
```diff
@@ -278,13 +278,17 @@ public void removeBlocks(List<Long> blockIds) {
           if (masterBlockInfo == null) {
             continue;
           }
+          ArrayList<Long> workersToRemove = new ArrayList<Long>();
           for (long workerId : masterBlockInfo.getWorkers()) {
-            masterBlockInfo.removeWorker(workerId);
+            workersToRemove.add(workerId);
```
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/block/BlockMaster.java]: What about this: for(long workerId : new ArrayList(masterBlockInfo.getWorkers()))?

Comment (calvinjia) [inline servers/src/main/java/tachyon/master/block/BlockMaster.java]: I'm not sure if that's any better than the current solution, I think either of these would need to be changed after we fix up what `MasterBlockInfo` provides. Do you have a strong preference for one implementation?

Comment (yupeng9) [inline servers/src/main/java/tachyon/master/block/BlockMaster.java]: I don't have strong preference, but the proposed one is simpler :)
But I agree it doesn't matter much here, if it'll be changed soon.

**Comment by yupeng9** [comment]
LGTM

File: servers/src/main/java/tachyon/master/block/BlockMaster.java (hunk 1 above)
Comment (calvinjia) [inline servers/src/main/java/tachyon/master/block/BlockMaster.java]: Oh nevermind, your way is much simpler, I've updated the implementation.

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Pull request merged**